src/bearbyWallet/BearbyAccount.ts

```diff
@@ -260,12 +260,13 @@ export class BearbyAccount implements Provider {
   public async deploySC(params: DeploySCParams): Promise<SmartContract> {
     try {
       const fee = Number(params.fee ?? (await this.minimalFee()));
-      const totalCost =
+      const maxCoins =
+        params.maxCoins ||
```

Comment:
use ?? instead of ||
if the user sets maxCoins to 0, it will be falsy so totalCoins will be taken.

with ?? it will do so only if maxCoins is undefined.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing